Update seastar for v25.3.x

BenPope opened the pull request:
Notes for reviewer:
* Many commits reordered compared to previous branches to colocate similar changes and fixups
* All the "old" OpenSSL work was ripped out, and replaced with https://github.com/scylladb/seastar/pull/2569
  * Moved some stuff after these commits to make future rebases easier, and keep the changes more in sync

pgellert commented on tests/unit/prometheus_test.cc:
```diff
@@ -0,0 +1,96 @@
+/*
```
These 3 commits (e9a912b6b98e99ec042e5e22d7f38357bede2ff0..75f1129b767e662a51cac24397e0eb94e687f6e0) can be dropped because they have been upstreamed (https://github.com/scylladb/seastar/pull/2400). The reason why they still show up as a diff is because the test was upstreamed under a different filename (prometheus_http_test.cc instead of prometheus_test.cc).

michael-redpanda reviewed (commented):
OpenSSL/TLS changes look good

BenPope commented:
Changes in [force-push:](https://github.com/redpanda-data/seastar/compare/58ce9185ec1d2a99a7b011e62796dd7b40c7f4e4..68c3a6f708759e73c86d7533afde9fbab9100e19)
* Added two most recent commits to v25.3.x-pre
* Removed 3 commits that were already upstream
* Deflake cpu_profiler test again (move from 5 allowed `dropped_samples` to 10)
